demo/repo/Jenkinsfile

```diff
@@ -17,26 +17,31 @@ parallel(longerTests: {
     runTests(servers, 20)
 })
 
-stage name: 'Staging', concurrency: 1
-node {
-    servers.deploy 'staging'
-}
+milestone ordinal: 1
+lock (resource: 'staging-server', inversePrecedence: true) {
+    stage name: 'Staging'
+    node {
+        servers.deploy 'staging'
+    }
 
-input message: "Does ${jettyUrl}staging/ look good?"
-try {
-    checkpoint('Before production')
-} catch (NoSuchMethodError _) {
-    echo 'Checkpoint feature available in CloudBees Jenkins Enterprise.'
+    input message: "Does ${jettyUrl}staging/ look good?"
+    try {
+        checkpoint('Before production')
+    } catch (NoSuchMethodError _) {
+        echo 'Checkpoint feature available in CloudBees Jenkins Enterprise.'
+    }
 }
 
-stage name: 'Production', concurrency: 1
-node {
-    sh "wget -O - -S ${jettyUrl}staging/"
-    echo 'Production server looks to be alive'
-    servers.deploy 'production'
-    echo "Deployed to ${jettyUrl}production/"
+milestone()
```

Comment:
Yes. I just wanted to show that it's possible to use mixed _defined_ and _non-defined_ ordinals in the same script (as it was kind of controversial in the `milestone` PR discussion).